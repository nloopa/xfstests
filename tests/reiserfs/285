#! /bin/bash
# FS QA Test No. 285
#
# SEEK_DATA/SEEK_HOLE sanity tests.
#
# Improved by Jeff.liu@oracle.com
# Creater: josef@redhat.com
#
#-----------------------------------------------------------------------
# Copyright (c) 2011 Oracle Inc.  All Rights Reserved.
# Copyright (c) 2011 Red Hat.  All Rights Reserved.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#-----------------------------------------------------------------------
#

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
status=1	# failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 15

# get standard environment, filters and checks
. ./common/rc
. ./common/filter

_supported_fs generic
_supported_os Linux

_require_seek_data_hole

BASE_TEST_FILE=$TEST_DIR/seek_sanity_testfile

[ -x $here/src/seek_sanity_test ] || _notrun "seek_sanitfy_tester not built"

# Disable extent zeroing for ext4 as that change where holes are created
if [ "$FSTYP" = "ext4" ]; then
	DEV=`basename $TEST_DEV`
	echo 0 >/sys/fs/ext4/$DEV/extent_max_zeroout_kb
fi

_cleanup()
{
	eval "rm -f $BASE_TEST_FILE.*"
}

$here/src/seek_sanity_test $BASE_TEST_FILE > $seqres.full 2>&1 ||
	_fail "seek sanity check failed!"

# success, all done
status=0
exit
