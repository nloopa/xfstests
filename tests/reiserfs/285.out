QA output created by 285
